File: POOVeiculos/PHP/Caminhao.php
<?php
require_once "Veiculo.php";

class Caminhao extends Veiculo {
    public function __construct($modelo, $tipo, $potencia, $ano) {
        parent::__construct($modelo, $tipo, $potencia, $ano);
    }

    public function __destruct() {
        echo "<p>Objeto Caminhão Destruído com Sucesso!<br/>";
    }
}

File: POOVeiculos/C++/Caminhao.cpp
#include "Caminhao.h"

Caminhao::Caminhao(string modelo, string tipo, float potencia, int ano) : Veiculo(modelo, tipo, potencia, ano) {

}

Caminhao::~Caminhao() {
    cout << "Objeto Caminhão Destruído com Sucesso!" << endl;
}
